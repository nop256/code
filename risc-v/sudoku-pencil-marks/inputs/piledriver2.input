....3.6.8.9...1....12.8......5...7.....146...68.....2.......973.6..1..4...9..7...
