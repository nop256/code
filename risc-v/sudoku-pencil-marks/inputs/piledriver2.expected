Please enter an 81-character board:

Testing pencil_marks with the following board:

+-----------------------+-----------------------+-----------------------+
| 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |       | 1 2 3 |       |
| 4 5 6   4 5 6   4 5 6 | 4 5 6     3     4 5 6 |   6     4 5 6     8   |
| 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |       | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6     9     4 5 6 | 4 5 6   4 5 6     1   | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 |       |       | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6     1       2   | 4 5 6     8     4 5 6 | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 |       |       | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 |
| 4 5 6   4 5 6     5   | 4 5 6   4 5 6   4 5 6 |   7     4 5 6   4 5 6 |
| 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 | 1 2 3 |       |       |       | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6   4 5 6   4 5 6 |   1       4       6   | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 | 7 8 9 | 7 8 9 |       |       |       | 7 8 9 | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |
|   6       8     4 5 6 | 4 5 6   4 5 6   4 5 6 | 4 5 6     2     4 5 6 |
|       |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       |       |       |
| 4 5 6   4 5 6   4 5 6 | 4 5 6   4 5 6   4 5 6 |   9       7       3   |
| 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       |       |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 |       | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 |       | 1 2 3 |
| 4 5 6     6     4 5 6 | 4 5 6     1     4 5 6 | 4 5 6     4     4 5 6 |
| 7 8 9 |       | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 |       | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6   4 5 6     9   | 4 5 6   4 5 6     7   | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |
+-----------------------+-----------------------+-----------------------+

The return value is 1
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
| . . . | . . . | . . . | . 2 . |       | . 2 . |       | 1 . . |       |
| 4 5 .   4 5 .   4 . . | 4 5 .     3     4 5 . |   6     . 5 .     8   |
| 7 . . | 7 . . | 7 . . | 7 . 9 |       | . . 9 |       | . . 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . 3 |       | . . 3 | . 2 . | . 2 . |       | . 2 3 | . . 3 | . 2 . |
| 4 5 .     9     4 . 6 | 4 5 6   . 5 6     1   | 4 5 .   . 5 .   4 5 . |
| 7 8 . |       | 7 8 . | 7 . . | 7 . . |       | . . . | . . . | 7 . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . 3 |       |       | . . . |       | . . . | . . 3 | . . 3 | . . . |
| 4 5 .     1       2   | 4 5 6     8     4 5 . | 4 5 .   . 5 .   4 5 . |
| 7 . . |       |       | 7 . 9 |       | . . 9 | . . . | . . 9 | 7 . 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 | . 2 3 |       | . 2 3 | . 2 . | . 2 3 |       | 1 . 3 | 1 . . |
| 4 . .   4 . .     5   | . . .   . . .   . . . |   7     . . 6   4 . 6 |
| . . 9 | . . . |       | . 8 9 | . . 9 | . 8 9 |       | . 8 9 | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . 2 3 | . 2 3 | . . 3 |       |       |       | . . 3 | . . 3 | . . . |
| . . .   . . .   . . . |   1       4       6   | . 5 .   . 5 .   . 5 . |
| 7 . 9 | 7 . . | 7 . . |       |       |       | . 8 . | . 8 9 | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | 1 . 3 | . . 3 | . . . | . . 3 | 1 . 3 |       | 1 . . |
|   6       8     4 . . | . 5 .   . 5 .   . 5 . | 4 5 .     2     4 5 . |
|       |       | 7 . . | 7 . 9 | 7 . 9 | . . 9 | . . . |       | . . 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 . | . 2 . | 1 . . | . 2 . | . 2 . | . 2 . |       |       |       |
| 4 5 .   4 5 .   4 . . | 4 5 6   . 5 6   4 5 . |   9       7       3   |
| . 8 . | . . . | . 8 . | . 8 . | . . . | . 8 . |       |       |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . 2 3 |       | . . 3 | . 2 3 |       | . 2 3 | . 2 . |       | . 2 . |
| . 5 .     6     . . . | . 5 .     1     . 5 . | . 5 .     4     . 5 . |
| 7 8 . |       | 7 8 . | . 8 9 |       | . 8 9 | . 8 . |       | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | . 2 3 |       | . 2 3 | . 2 . |       | 1 2 . | 1 . . | 1 2 . |
| 4 5 .   4 5 .     9   | 4 5 6   . 5 6     7   | . 5 .   . 5 6   . 5 6 |
| . 8 . | . . . |       | . 8 . | . . . |       | . 8 . | . 8 . | . . . |
+-----------------------+-----------------------+-----------------------+

The return value is 0
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
| . . . | . . . | . . . | . 2 . |       | . 2 . |       | 1 . . |       |
| 4 5 .   4 5 .   4 . . | 4 5 .     3     4 5 . |   6     . 5 .     8   |
| 7 . . | 7 . . | 7 . . | 7 . 9 |       | . . 9 |       | . . 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . 3 |       | . . 3 | . 2 . | . 2 . |       | . 2 3 | . . 3 | . 2 . |
| 4 5 .     9     4 . 6 | 4 5 6   . 5 6     1   | 4 5 .   . 5 .   4 5 . |
| 7 8 . |       | 7 8 . | 7 . . | 7 . . |       | . . . | . . . | 7 . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . 3 |       |       | . . . |       | . . . | . . 3 | . . 3 | . . . |
| 4 5 .     1       2   | 4 5 6     8     4 5 . | 4 5 .   . 5 .   4 5 . |
| 7 . . |       |       | 7 . 9 |       | . . 9 | . . . | . . 9 | 7 . 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 | . 2 3 |       | . 2 3 | . 2 . | . 2 3 |       | 1 . 3 | 1 . . |
| 4 . .   4 . .     5   | . . .   . . .   . . . |   7     . . 6   4 . 6 |
| . . 9 | . . . |       | . 8 9 | . . 9 | . 8 9 |       | . 8 9 | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . 2 3 | . 2 3 | . . 3 |       |       |       | . . 3 | . . 3 | . . . |
| . . .   . . .   . . . |   1       4       6   | . 5 .   . 5 .   . 5 . |
| 7 . 9 | 7 . . | 7 . . |       |       |       | . 8 . | . 8 9 | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | 1 . 3 | . . 3 | . . . | . . 3 | 1 . 3 |       | 1 . . |
|   6       8     4 . . | . 5 .   . 5 .   . 5 . | 4 5 .     2     4 5 . |
|       |       | 7 . . | 7 . 9 | 7 . 9 | . . 9 | . . . |       | . . 9 |
+-----------------------+-----------------------+-----------------------+
| 1 2 . | . 2 . | 1 . . | . 2 . | . 2 . | . 2 . |       |       |       |
| 4 5 .   4 5 .   4 . . | 4 5 6   . 5 6   4 5 . |   9       7       3   |
| . 8 . | . . . | . 8 . | . 8 . | . . . | . 8 . |       |       |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . 2 3 |       | . . 3 | . 2 3 |       | . 2 3 | . 2 . |       | . 2 . |
| . 5 .     6     . . . | . 5 .     1     . 5 . | . 5 .     4     . 5 . |
| 7 8 . |       | 7 8 . | . 8 9 |       | . 8 9 | . 8 . |       | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | . 2 3 |       | . 2 3 | . 2 . |       | 1 2 . | 1 . . | 1 2 . |
| 4 5 .   4 5 .     9   | 4 5 6   . 5 6     7   | . 5 .   . 5 6   . 5 6 |
| . 8 . | . . . |       | . 8 . | . . . |       | . 8 . | . 8 . | . . . |
+-----------------------+-----------------------+-----------------------+
